### ISSUE TITLE
Simplify XML description of model variables

### ISSUE THREAD

**Issue by t-sommer** [modelica/fmi-standard#514]
Currently the type specific information of the variables is stored in a nested tag inside the `<Variable>` tag e.g.

```
<ModelVariables>
  <Variable name="v" valueReference="1" description="Velocity">
    <Float64 start="0.0" unit="m/s"/>
  </Variable>
  <Variable name="gear" valueReference="2" description="Gear number">
    <Int32 start="1"/>
  </Variable>
</ModelVariables>
```

However the same information could be stored as

```
<ModelVariables>
  <Float64 name="v" valueReference="1" description="Velocity" start="0.0" unit="m/s" />
  <Int32 name="gear" valueReference="2" description="Gear number" start="1" />
</ModelVariables>
```

which results in a more compact XML and provides better readability.

**Comment by KarlWernersson** [comment]
In fmi 2 we have 7 attributes that are common for all variable types and they are stored in scalarSariable
Then the scalar variable has a one type node that contians type specific attributes which varies in number from 12 in Real, and 2 in Boolean and String. As you propose this can be flattened in wich you would get for Fmi 3.0 11 (if I counted right) different kinds of variables, instead of one variables with 11 different kinds of types.

I don't currently see any great benefit of doing it one way or another, its more a matter of taste. But since we have had it hierarchical in FMI 1.0 & 2.0 I would prefer not to change it unless we have a good motivation for it.

**Comment by TorstenBlochwitz** [comment]
I want to discuss Masoud's remark form above: 
Masoud: the old definition has the advantage that in the future we could define structures
Andreas J: Not with me being around

Within the EMPHYSIS project we currently specify eFMI (this is a working title), which will allow the direct integration of physical models as production code into ECUs for model based control and diagnosis applications. We try to reuse as much as possible from FMI.
In eFMI we have to deal with structured data. For data exchange we don't use the fmiXGet/Set functions. Data exchange is organized as it is usual for embedded systems (global variables, access macros, access functions, ...). For that reasons we do not run into the complexity problem of Hilding's proposal some years ago, which tried to use general get/set functions for the direct access of structured data.
In eFMI, as a way out for structured data, we could introduce an additional element in parallel to the above mentioned elements (Float64, Int32… ) and call it for example “Structured”. This element could have its member variables as children.
In this way (from my point of view), Torsten Sommer’s proposal would also work for structured variables.

**Comment by TorstenBlochwitz** [comment]
Some people in the eFMI specification group asked about the way how attributes for variables (start, min, max) are defined. In FMI 2.0 we use attributes of different types in order to be consistent with the type of the variable itself. Jörg Niere (dSPACE) came up with the idea to use Strings for that. 
With the current proposal, discussed in this thread, the usage of typed attributes is obvious. But Jörg mentioned one advantage of using strings to express floating point variables.
@Jörg: could you please elaborate this a bit more in detail?

**Comment by t-sommer** [comment]
@TorstenBlochwitz, can you open a new issue for this "Structured" variable? It's a new type and (almost) unrelated to the simplification of the XML structure.

**Comment by t-sommer** [comment]
I've created an [updated schema](https://github.com/t-sommer/fmi-standard/tree/variable-inheritance) that reflects the proposed changes. @pmai, can you take a look at it and let me know if this is the right way to implement it in the XSD?

**Comment by pmai** [comment]
I've looked at this and it seems like a good start. I've added the base derivation to the other variables and refactored it a bit to keep the expected structure; could you give me push rights to your branch?

If the declaredType attribute was not required for enums, we could merge that into the base type, however I currently do not see how to achieve this with XML Schema 1.0 (or 1.1) for that matter, except maybe with extra assertions.

Other than that, one might want to think about the type declaration stuff, I'll give this a think on the plane... ;).

**Comment by t-sommer** [comment]
I've just added you as a contributor, @pmai. I'll also update the examples so it passes CI.

**Comment by chrbertsch** [comment]
As #530 has been merged: can this be closed?

**Comment by klausschuch** [comment]
The merged changes are only half the story, the specification document must be updated too.
The specification document references the ScalarVariable (or Variable) xml tag in several places.

**Comment by t-sommer** [comment]
Should we use `<Variable>` as a placeholder for the actual XML elements (`<Int32>`, `<Float64>`, ...)?

**Comment by chrbertsch** [comment]
Could be part of a "Write Session" by 2-3 people.
Torsten Sommer asks for volunteers. (Klaus and Christian B. would be willing, it is a question of time ...)

**Comment by t-sommer** [comment]
Solved with #530.
